### PR TITLE
transitions: override the method is_same_location

### PR DESCRIPTION
**Pull request by BadrAly**
Co-Authored-by: Aly Badr <aly.badr@rero.ch>

### CODE REVIEW

File: invenio_circulation/utils.py [hunk 1]
```diff
@@ -168,3 +168,18 @@ def validate_item_pickup_transaction_locations(loan, destination, **kwargs):
...
+    """Validates location of given item_pid and given location are the same.
+
+    This also allow for extra validation if needed at the library level.
+
+    :param item_pid: a dict containing `value` and `type` fields to
+        uniquely identify the item.
+    :param location_pid: a location pid.
+    :return: False if validation is not possible, otherwise True
+    """
+    raise NotImplementedConfigurationError(
```
Comment (ntarocco) [inline invenio_circulation/utils.py]: Do we want to have a not implemented version by default, or keep the previous one as default?
```
    item_location_pid = current_app.config["CIRCULATION_ITEM_LOCATION_RETRIEVER"](item_pid)
    return input_location_pid == item_location_pid
```

Comment (BadrAly) [reply invenio_circulation/utils.py]: done

**Review by ntarocco — APPROVED** [review]

File: invenio_circulation/transitions/transitions.py [hunk 2]
```diff
@@ -95,16 +94,23 @@ def _update_document_pending_request_for_item(item_pid, **kwargs):
         current_circulation.loan_indexer().index(pending_loan)
 
 
+def _is_same_location(item_pid, location_pid):
+    """Validates location of given item_pid and given location are the same."""
```
Comment (jma) [inline invenio_circulation/transitions/transitions.py]: params are missing in the docstring

File: tests/utils.py [hunk 3]
```diff
@@ -99,3 +99,9 @@ def validate_item_pickup_transaction_locations(loan, destination, **kwargs):
         return pickup_location_pid == item_location_pid
     elif destination == "ITEM_IN_TRANSIT_FOR_PICKUP":
         return pickup_location_pid != item_location_pid
+
+
+def same_location_validator(item_pid, location_pid):
+    """Validates location of given item_pid and given location are the same."""
```
Comment (jma) [inline tests/utils.py]: params and return are missing in docstring.

**Review by jma — APPROVED** [review]

**Review by kpsherva — APPROVED** [review]

**Pull request merged**